app_server/app/controllers/StreamController.js

```diff
@@ -0,0 +1,143 @@
+/**
+ * Stream Controller
+ * @module StreamController
+ */
+var rfr = require('rfr');
+var Joi = require('joi');
+var Boom = require('boom');
+var Promise = require('bluebird');
+var crypto = require('crypto');
+var hash = crypto.createHash('sha256');
+
+var CustomError = rfr('app/util/Error');
+var Utility = rfr('app/util/Utility');
+var Service = rfr('app/services/Service');
+var Authenticator = rfr('app/policies/Authenticator');
+
+var logger = Utility.createLogger(__filename);
+
+function StreamController(server, options) {
+  this.server = server;
+  this.options = options;
+}
+
+var Class = StreamController.prototype;
+
+Class.registerRoutes = function () {
+  this.server.route({method: 'GET', path: '/',
+                     config: {
+                       validate: streamListParamsValidator,
+                       auth: {scope: Authenticator.SCOPE.ALL}
+                     },
+                     handler: this.getListOfStreams});
+
+  this.server.route({method: 'GET', path: '/{id}',
+                     config: {
+                       validate: singleStreamValidator,
+                       auth: {scope: Authenticator.SCOPE.ALL}
+                     },
+                     handler: this.getStreamById});
+
+  this.server.route({method: 'POST', path: '/',
+                     config: {
+                       validate: streamCreatePayloadValidator,
+                       auth: {scope: Authenticator.SCOPE.ALL}
+                     },
+                     handler: this.createStream});
+};
+
+/* Routes handlers */
+Class.createStream = function (request, reply) {
+  logger.debug('Creating stream');
+
+  var userId = request.auth.credentials.userId;
+  var currTime = Date.now();
+
+  return Promise.method(function generateStream() {
+    var newStream = {
+      title: request.payload.title,
+      description: request.payload.description,
+      createdAt: currTime,
+      appInstance: hash.update(userId + currTime).digest('hex')
+    };
```

Comment:
This `hash` object will be invalidated after the call to `digest` as documented [here](https://nodejs.org/api/crypto.html#crypto_hash_digest_encoding). Did I miss any other details since this could make this `hash` unusable when creating a 2nd stream?